read-fonts/src/tables/gsub/closure.rs

```diff
@@ -0,0 +1,188 @@
+//! Computing the closure over a set of glyphs
+//!
+//! This means taking a set of glyphs and finding the set of glyphs that are
+//! reachable from those glyphs via substitution.
+
+use std::collections::HashSet;
+
+use font_types::GlyphId;
+
+use crate::ReadError;
+
+use super::{
+    AlternateSubstFormat1, Gsub, LigatureSubstFormat1, MultipleSubstFormat1, SingleSubst,
+    SingleSubstFormat1, SingleSubstFormat2, SubstitutionSubtables,
+};
+
+#[cfg(feature = "std")]
+impl<'a> Gsub<'a> {
+    /// Return the set of glyphs reachable from the input set via any substituion.
+    pub fn closure_glyphs(&self, glyphs: HashSet<GlyphId>) -> Result<HashSet<GlyphId>, ReadError> {
```

Comment:
this could potentially become part of a trait when we will have to implement closure_glyphs for more tables other than GSUB (cmap, glyf, COLR, MATH, etc.)